code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -1336,3 +1336,837 @@
+				src.adjustHalLoss(20)
+				src.flash_pain(20)
+				host.last_action = world.time + 10 MINUTES
+				return
+		else if(S.shielded == SOCKET_SHIELDED)
+			if(prob(50))
+				to_chat(src, SPAN_WARNING("You feel a sudden pain in your neural socket as you are repelled by [target]'s countermeasures!"))
+				to_chat(target, SPAN_WARNING("[src] attempted to ban you from the Hivenet, but your countermeasures repelled them!"))
+				src.adjustHalLoss(15)
+				src.flash_pain(15)
+				host.last_action = world.time + 10 MINUTES
+				return
+			else
+				to_chat(src, SPAN_NOTICE("You extend your will, severing [target]'s connection to the Hivenet. Perhaps now it will learn its lesson."))
+				to_chat(target, SPAN_WARNING("You feel the thoughts of your fellow Vaurcae abruptly vanish, as [src]'s will smothers them all. In an instant, your neural socket is banned, cut off from hearing or speaking to the wider Hivenet."))
+				target.remove_language(LANGUAGE_VAURCA)
+				S.banned = TRUE
+				host.last_action = world.time + 1 MINUTES
+		else if(S.shielded == SOCKET_FULLSHIELDED)
+			to_chat(src, SPAN_WARNING("You feel a sudden pain in your neural socket as you are repelled by [target]'s countermeasures!"))
+			to_chat(target, SPAN_WARNING("[src] attempted to ban you from the Hivenet, but your countermeasures repelled them!"))
+			src.adjustHalLoss(15)
+			src.flash_pain(15)
+			host.last_action = world.time + 10 MINUTES
+			return
+
+/mob/living/carbon/human/proc/hivevoid() //Destroys a Vaurca's neural socket.
+	set name = "Void Hivenet User"
+	set desc = "Permanently sever a Vaurca of your own Hive from the Hivenet, destroying their neural socket. Mouv Ta may attempt to use this on Vaurca of another Hive, though this may have severe diplomatic consequences!"
+	set category = "Hivenet"
+	var/list/available_vaurca = list()
+	var/list/fullname = splittext(src.name, " ")
+	var/surname = fullname[2]
+	var/obj/item/organ/internal/vaurca/neuralsocket/host = src.internal_organs_by_name[BP_NEURAL_SOCKET]
+	if(src.stat != CONSCIOUS)
+		to_chat(src, SPAN_WARNING("You are incapable of that in your current state!"))
+		return
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !istype(host))
+		to_chat(src, "<span class='danger'>Your mind is dark, unable to communicate with the Hive.</span>")
+		return
+	if(!istype(src.internal_organs_by_name[BP_NEURAL_SOCKET], /obj/item/organ/internal/vaurca/neuralsocket/admin))
+		to_chat(src, "<span class='warning'>You lack the authority to do this!.</span>")
+		return
+	for(var/mob/living/carbon/human/player in player_list)
+		if(isvaurca(player) && player.internal_organs_by_name[BP_NEURAL_SOCKET])
+			var/list/player_fullname = splittext(player.name, " ")
+			var/player_surname = player_fullname[2]
+			if(player_surname == surname || istype(src.origin, /singleton/origin_item/origin/mouv_b))
+				LAZYADD(available_vaurca, player)
+	LAZYREMOVE(available_vaurca, src) //can't ban/mute/void yourself
```

Comment:
not a lazylist, don't use LAZY macros

`for(var/mob/living/carbon/human/player in (player_list - src))` should work

also, shouldn't this be human_mob_list instead?